tsconfig.json

```diff
@@ -1,15 +1,12 @@
 {
   "include": ["*.js"],
   "compilerOptions": {
-    "target": "ES2020",
-    "lib": ["ES2020"],
-    "module": "ES2020",
-    "moduleResolution": "node",
-    "allowJs": true,
+    "target": "es2021",
+    "lib": ["es2021"],
+    "module": "node16",
     "checkJs": true,
     "declaration": true,
     "emitDeclarationOnly": true,
-    "allowSyntheticDefaultImports": true,
     "skipLibCheck": true,
```

Comment:
ideally we could also remove `skilLibCheck`
Currently remaining:
```
node_modules/rehype-stringify/lib/index.d.ts:3:25 - error TS2307: Cannot find module 'hast-util-to-html/lib/types' or its corresponding type declarations.

3   config: void | import('hast-util-to-html/lib/types').Options | undefined
                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

node_modules/remark-parse/lib/index.d.ts:3:26 - error TS2307: Cannot find module 'mdast-util-from-markdown/lib' or its corresponding type declarations.

3   options: void | import('mdast-util-from-markdown/lib').Options | undefined
```

these should be solved by https://github.com/rehypejs/rehype/pull/123 and https://github.com/remarkjs/remark/pull/1106 🤞